modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/GridCacheDatabaseSharedManager.java

```diff
@@ -1772,9 +1781,60 @@ private boolean safeToUpdatePageMemories() {
             }
         }
 
+        if (log.isInfoEnabled() && !F.isEmpty(earliestValidCheckpoints))
+            printReservationToLog(earliestValidCheckpoints);
+
         return grpPartsWithCnts;
     }
 
+    /**
+     * Prints detail information about caches which were not reserved
+     * and reservation depth for the caches which have WAL history enough.
+     *
+     * @param earliestValidCheckpoints Map contains information about caches' reservation.
+     */
+    private void printReservationToLog(
+        Map<Integer, T2<ReservationReason, Map<Integer, CheckpointEntry>>> earliestValidCheckpoints) {
+
+        Map<ReservationReason, List<Integer>> notReservedCachesToPrint = new HashMap<>();
+        Map<ReservationReason, List<T2<Integer, CheckpointEntry>>> reservedCachesToPrint = new HashMap<>();
+
+        for (Map.Entry<Integer, T2<ReservationReason, Map<Integer, CheckpointEntry>>> entry : earliestValidCheckpoints.entrySet()) {
+            if (entry.getValue().get2() == null) {
+                notReservedCachesToPrint.computeIfAbsent(entry.getValue().get1(), reason -> new ArrayList<>())
+                    .add(entry.getKey());
+            }
+            else {
+                reservedCachesToPrint.computeIfAbsent(entry.getValue().get1(), reason -> new ArrayList<>())
+                    .add(new T2(entry.getKey(), entry.getValue().get2().values().stream().min((cp1, cp2) ->
+                        Long.compare(cp1.timestamp(), cp2.timestamp())).get()));
+            }
+        }
+
+        if (!F.isEmpty(notReservedCachesToPrint)) {
+            log.info("Following caches were not reserved [" +
+                notReservedCachesToPrint.entrySet().stream()
+                    .map(entry -> '[' +
+                        entry.getValue().stream().map(grpId -> "[grpId=" + grpId +
+                            ", grpName=" + cctx.cache().cacheGroup(grpId).cacheOrGroupName() + ']')
+                            .collect(Collectors.joining(", ")) +
+                        ", reason=" + entry.getKey() + ']')
+                    .collect(Collectors.joining(", ")) + ']');
+        }
+
+        if (!F.isEmpty(reservedCachesToPrint)) {
+            log.info("Reserved cache groups with first reserved checkpoint IDs and reasons why previous checkpoint was inapplicable: [" +
```

Comment:
Reserved cache groups with first reserved checkpoint IDs and reasons why previous checkpoint was inapplicable
->
Cache groups with earliest reserved checkpoint and a reason why a previous checkpoint was inapplicable

Reply:
Done.